recipes/msdfgen/all/conanfile.py

```diff
@@ -29,85 +34,94 @@ class MsdfgenConan(ConanFile):
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("freetype/2.11.1")
+        self.requires("freetype/2.12.1")
         self.requires("lodepng/cci.20200615")
         self.requires("tinyxml2/9.0.0")
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
-        if self._is_msvc and self.options.shared:
-            raise ConanInvalidConfiguration("msdfgen shared not supported by Visual Studio")
-        if self.options.with_skia:
+        if self.info.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 11)
+        if is_msvc(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} shared not supported by Visual Studio")
+        if self.info.options.with_skia:
             raise ConanInvalidConfiguration("skia recipe not available yet in CCI")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["MSDFGEN_BUILD_MSDFGEN_STANDALONE"] = self.options.utility
+        tc.variables["MSDFGEN_USE_OPENMP"] = self.options.with_openmp
+        tc.variables["MSDFGEN_USE_CPP11"] = True
+        tc.variables["MSDFGEN_USE_SKIA"] = self.options.with_skia
+        tc.variables["MSDFGEN_INSTALL"] = True
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        apply_conandata_patches(self)
+        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
+        # workaround against CMAKE_FIND_PACKAGE_PREFER_CONFIG ON in conan toolchain
+        replace_in_file(self, cmakelists, "find_package(Freetype REQUIRED)", "find_package(Freetype REQUIRED MODULE)")
```

Comment:
Can you pretty please move some of these to patches?

I'd really love to have the patch type so we can track the changes we need to make 🙏

Reply:
it's a patch we have to keep forever, so it's not a good idea to move it to a patch file.

Reply:
They are difficult to review, we can not easily see what effect its having and what changes are being made. They are also difficult to track and understand for consumers.

We are starting to have the conversation about security in CCI so we may start adding policies in place to improve that.


For example removing inactive users from the access list. Is a recent thing.


I understand you point about maintaining patches but security is more important, I think most people would agree